### PR TITLE
Add script for formatting icon data

### PR DESCRIPTION
**Pull request by LitoMore**
This script can help users automatically fix the `simple-icons.json` order and final newline.

I also fixed the compare function for two icons that have the same title, like `Hive (hive)` and `Hive (hive_blockchain)`.

### CODE REVIEW

File: scripts/format-icon-data.js
Comment (uncenter) [inline scripts/format-icon-data.js:1]: "format" seems misleading, I would use "sort". This would be `sort-icon-data.js`.

File: package.json [hunk 1]
```diff
@@ -113,7 +113,8 @@
   "scripts": {
     "build": "node scripts/build/package.js",
     "clean": "node scripts/build/clean.js",
-    "format": "npm run prettier -- --write && npm run xo:fix",
+    "format:icondata": "node scripts/format-icon-data.js",
+    "format": "npm run format:icondata && npm run prettier -- --write && npm run xo:fix",
```
Comment (uncenter) [inline package.json:117]: What's the benefit in splitting the icon data into its own script here? I can't see myself ever using it separately from format.

Comment (LitoMore) [reply package.json:117]: To make our scripts consistent. Like `xo:fix` does. The change was requested by @mondeja.

File: scripts/format-icon-data.js
Comment (LitoMore) [reply scripts/format-icon-data.js:1]: This also reformats the file. Even you use other indentations.

**Review by mondeja — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (mondeja) [reply package.json:117]: When you separate npm scripts and run them from another one, npm shows in their output each `npm run` command executed. If a failure happens or the command takes some apreciable time, it's easy to know what script is causing it.

File: scripts/format-icon-data.js
Comment (mondeja) [reply scripts/format-icon-data.js:1]: I agree with @LitoMore here, sorting is something included in the formatting process.

**Pull request merged**